class-lab/exercise5-autoComplete/AutoCompleteTextBox.js

```diff
@@ -0,0 +1,54 @@
+class AutoCompleteTextBox {
+  constructor(el) {
+    this.data = ["1111", "2222", "3333", "4444", "5555", "6666", "7777", "8888",
+      "9999", "129", "a38", "aaa", "12345", "321", "b38"];
+    el.innerHTML = `
+      <div>
+          <datalist id="suggestions"></datalist>
+          <input autoComplete="of" list="suggestions" type="search" placeholder="Search Google or type a URL"/> 
+      </div>`
+    this.suggestions = el.querySelector('datalist');
+    this.input = el.querySelector('input');
+    this.input.addEventListener('input', this.autoComplete);
+  }
+
+  autoComplete = () => {
+    const stringToSearch = this.input.value;
+    let searchResult = this.search(stringToSearch);
+    console.log("searchResult " + searchResult);
+    this.addOptions(searchResult);
+  }
+
+  addOptions = (searchResult) => {
+    this.suggestions.innerHTML = "";
+    for (let item of searchResult) {
+      const option = document.createElement('option');
```

Comment:
You are misusing the `datalist` attribute. You are populating and repopulating it on each search. It should have constant data. Look at how it works here https://developer.mozilla.org/en-US/docs/Web/HTML/Element/datalist

Reply:
but the data in the datalist should change according to the user input, this is different than the link you sent.

Reply:
Yes, because the link I sent is how `datalist` **should** be used. Seems like it is not the correct element for your case since its default behavior is not what you want. Try to find a solution using another element.